src/service/engine/index.ts

```diff
@@ -90,41 +96,60 @@ export class EngineService {
     return engines;
   }
 
+  private async validateCreateOptions(
+    accountVersion: number,
+    options: CreateEngineOptions
+  ) {
+    const disallowedV2Options = ["region", "engine_type", "warmup"];
```

Comment:
I'm thinking, can we instead of having a list of disallowed options have a v2 type instead? Then we'll have the create function signature contain:
```
options:  CreateEngineOptionsV1 | CreateEngineOptionsV2 = {}
```
and check that account version matches the parameter type "class" passed.

Reply:
I think that the problem here would be that CreateEngineOptionsV2 would be a subclass of CreateEngineOptionsV1, therefore any `options` value is always a `CreateEngineOptionsV1`. Yet I'll still try this approach and see it will result in something clear

Reply:
Tested this approach, and it seems a bit overcomplicated. A couple of cons
- we are not able to tell users which options are not supported anymore (which might make it harder to migrate)
- it slightly breaks backward compatilibity, since we have different signatures for v1 and v2
I believe we can have this logic while v1 is still there, so we can give users better errors for deprecated parameters, and we'll refactor this into a propper typing later